app/components/Header/ToggleTheme.tsx

```diff
@@ -32,6 +33,10 @@ const ToggleTheme = ({
     setIcon(getIcon());
   }, []);
 
+  window.addEventListener('themeChange', () => {
+    setIcon(getIcon());
+  });
+
```

Comment:
This will unfortunately have some unintended consequences (server-side rendering will break & it will register multiple eventListeners on every re-render).

If you move this into the `useEffect` above, and add a [cleanup function](https://react.dev/reference/react/useEffect#examples-connecting) to remove the listener it should work fine though :)

Reply:
Thanks for the quick review! Should be done now! 😃